### PR TITLE
#30 - Github action for binary release workflow

### PR DESCRIPTION
**Pull request by mtps**
## Description

relates to: #30 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
  - Not applicable: change was adding github actions.
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
  - Not applicable: no go code was changed.
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
  - Not applicable: no go code was changed.
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by dpederson-figure — APPROVED** [review]
lgtm

File: .github/workflows/matrix.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        id: create_release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ github.ref }}
+          draft: true
+          prerelease: true
+          body: |
+            ### Features
```
Comment (iramiller) [inline .github/workflows/matrix.yml:63]: is it possible to pull in the RELEASE_CHANGELOG.md file contents for this part?

**Review by iramiller — COMMENTED** [review]
Good looking PR.  Question about the release notes though and if we can get the RELEASE_CHANGELOG contents for that or not.

**Comment by iramiller** [comment]
Might be nice to have our change logs linked up by running a script over them
```
import fileinput
import re

# This script goes through the provided file, and replaces any " \#<number>",
# with the valid mark down formatted link to it. e.g.
# " [\#number](https://github.com/cosmos/cosmos-sdk/issues/<number>)
# Note that if the number is for a PR, github will auto-redirect you when you click the link.
# It is safe to run the script multiple times in succession. 
#
# Example:
#
# $ python ./scripts/linkify_changelog.py CHANGELOG.md
for line in fileinput.input(inplace=1):
    line = re.sub(r"\s\\#([0-9]+)", r" [\\#\1](https://github.com/provenance-io/provenance/issues/\1)", line.rstrip())
    print(line)
```

**Review by iramiller — DISMISSED** [review]
LGTM

**Comment by mtps** [comment]
@iramiller i think we are good to merge this now.

**Comment by mtps** [comment]
@iramiller i created a new issue to clean up the changelog links

**Review by dpederson-figure — APPROVED** [review]

**Review by iramiller — APPROVED** [review]
Still LGTM

**Pull request merged**